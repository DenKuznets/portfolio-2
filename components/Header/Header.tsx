'use client';
import Image from 'next/image';
import logoIcon from '../../public/images/logo.png';
import NavBar from '../NavBar/NavBar';
import Link from 'next/link';
import { useState } from 'react';
import Burger from '../Burger/Burger';

const Header = () => {
    const [showMenu, setShowMenu] = useState(false);
    return (
        <header
            data-testid={'header'}
            className="w-full top-0 z-10 shadow-md md:fixed md:p-4 flex justify-between p-2 bg-white"
        >
            <Link
                className="flex items-center text-inherit hover:no-underline"
                href="#"
            >
                <Image
                    className="mr-2 h-full w-auto"
                    sizes="(max-width: 768px) 20px, 30px"
                    src={logoIcon}
                    alt="html5 and hammer image"
                />
                <span className="font-bold text-sm sm:text-lg whitespace-nowrap">
                    ｄｅｎ . ｋｕｚｎｅｔｓ
                </span>
            </Link>

            <NavBar
                handleClick={() => setShowMenu(!showMenu)}
                className={`transition-all duration-200 ${
                    showMenu ? 'left-0 top-0' : 'left-[-100vh] '
                }`}
            />

            <Burger
                className="fixed right-6 top-4 sm:top-5 md:hidden z-50 "
                isOpen={showMenu}
                handleClick={() => setShowMenu(!showMenu)}
            />
        </header>
    );
};

export default Header;
