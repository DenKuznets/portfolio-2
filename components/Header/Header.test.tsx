import { render, screen } from "@testing-library/react";
import Header from "./Header";
import "@testing-library/jest-dom";

test("renders correctly", () => {
  render(<Header />);
  const logoImage = screen.getByAltText('html5 and hammer image');
  const navbar = screen.getByRole('navigation');
  expect(logoImage).toBeInTheDocument();
  expect(navbar).toBeInTheDocument();
});

// test("renders HEADER unchanged", () => {
//   const { container } = render(<Header />);
//   expect(container).toMatchSnapshot();
// });
