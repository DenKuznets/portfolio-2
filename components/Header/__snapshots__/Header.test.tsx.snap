// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header unchanged 1`] = `
<div>
  <header
    class="w-full top-0 z-10 shadow-md md:fixed md:p-4 flex justify-between p-2 bg-white"
    data-testid="header"
  >
    <a
      class="flex items-center text-inherit hover:no-underline"
      href="#"
    >
      <img
        alt="html5 and hammer image"
        class="mr-2 h-full w-auto"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        sizes="(max-width: 768px) 5vw, 8vw"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 32w, /_next/image?url=%2Fimg.jpg&w=48&q=75 48w, /_next/image?url=%2Fimg.jpg&w=64&q=75 64w, /_next/image?url=%2Fimg.jpg&w=96&q=75 96w, /_next/image?url=%2Fimg.jpg&w=128&q=75 128w, /_next/image?url=%2Fimg.jpg&w=256&q=75 256w, /_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="color: transparent;"
        width="40"
      />
      <span
        class="font-bold text-sm sm:text-lg whitespace-nowrap"
      >
        ｄｅｎ . ｋｕｚｎｅｔｓ
      </span>
    </a>
    <nav
      class="max-md:hidden "
    >
      <ul
        class="flex items-center justify-center gap-y-10 text-center font-serif max-md:min-h-screen max-md:flex-col max-md:text-[2rem] md:gap-x-16"
      >
        <li>
          <a
            class="text-gray-500 hover:text-almostBlack"
            href="#hero"
          >
            Обо мне
          </a>
        </li>
        <li>
          <a
            class="text-gray-500 hover:text-almostBlack"
            href="#examples"
          >
            Примеры работ
          </a>
        </li>
      </ul>
    </nav>
    <button
      aria-label="Hamburger Menu"
      class="fixed right-6 top-4 sm:hidden z-50 "
    >
      <div
        class="w-6 cursor-pointer transition-all ease duration-300"
      >
        <div
          class="relative"
        >
          <span
            class="absolute top-0 inline-block w-full h-0.5 bg-almostBlack rounded transition-all ease duration-200"
            style="transform: rotate(0deg) translateY(6px);"
          >
             
          </span>
          <span
            class="absolute top-0 inline-block w-full h-0.5 bg-almostBlack rounded transition-all ease duration-200"
            style="opacity: 1;"
          >
             
          </span>
          <span
            class="absolute top-0 inline-block w-full h-0.5 bg-almostBlack rounded transition-all ease duration-200"
            style="transform: rotate(0deg) translateY(-6px);"
          >
             
          </span>
        </div>
      </div>
    </button>
  </header>
</div>
`;
